frontend/pages/software/ManageSoftwarePage/ManageSoftwarePage.tsx

```diff
@@ -613,12 +613,64 @@ const ManageSoftwarePage = ({
       searchQuery !== "" ||
       queryParams.vulnerable === "true");
 
-  return !isRouteOk ||
-    (isPremiumTier && !userTeams) ||
-    !globalConfig ||
-    (!softwareConfig && !softwareConfigError) ? (
-    <Spinner />
-  ) : (
+  const renderSoftwareTable = () => {
+    if (
+      isFetchingCount ||
+      isFetchingSoftware ||
+      !globalConfig ||
+      (!softwareConfig && !softwareConfigError)
+    ) {
+      return <Spinner />;
```

Comment:
To add a note here – we are gating the rendering TableContainer here with these conditions instead of determining the `isLoading` prop on TableContainer with the same conditions to make sure that TableContainer doesn't begin to render before all of the conditions are satisfied, per @gillespi314's comment – using the `isLoading` prop doesn't guarantee that